src/Integration.Vsix/Suppression/LiveIssueFactory.cs

```diff
@@ -0,0 +1,143 @@
+    internal sealed class LiveIssueFactory
+    {
+        private readonly Workspace workspace;
+        private readonly IServiceProvider serviceProvider;
+
+        /// <summary>
+        /// Mapping from full project file path to the unique project id
+        /// </summary>
+        private Dictionary<string, string> projectPathToProjectIdMap;
+
+        public LiveIssueFactory(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+
+            var componentModel = (IComponentModel)serviceProvider.GetService(typeof(SComponentModel));
+            workspace = componentModel.GetService<VisualStudioWorkspace>();
+
+            BuildProjectPathToIdMap();
+        }
+
+        /// <summary>
+        /// Attempts to fetch the extra information required to map from a Roslyn issue
+        /// to a SonarQube server issue. Returns null if there is not enough information
+        /// to produce the mapping.
+        /// </summary>
+        public LiveIssue Create(Diagnostic diagnostic)
+        {
+            // Get the file and project containing the issue
+            SyntaxTree tree = diagnostic.Location?.SourceTree;
+            if (tree == null)
+            {
+                return null;
+            }
+
+            if (diagnostic.Location == Location.None)
+            {
+                return null;
+            }
+
+            Project project = workspace?.CurrentSolution?.GetDocument(tree)?.Project;
+            if (project == null)
+            {
+                return null;
+            }
+
+            string projectId;
+            if (!projectPathToProjectIdMap.TryGetValue(project.FilePath, out projectId))
+            {
+                Debug.Fail($"Expecting to have a project id for the Roslyn project: {project.FilePath}");
+                return null;
+            }
+
+            // Get the whole of the line of text containing the issue (needed to compute the checksum)
+            // TODO: checksum for file level issues?
+            FileLinePositionSpan lineSpan = diagnostic.Location.GetLineSpan();
+
+            int startLine = lineSpan.StartLinePosition.Line;
+            int additionalLineCount = lineSpan.EndLinePosition.Line - startLine;
+            string lineText = tree.GetText().Lines[startLine + additionalLineCount].ToString();
+
+            LiveIssue liveIssue = new LiveIssue(diagnostic, lineSpan.Path, project.FilePath, projectId,
+                startLine + 1, lineText); // Roslyn lines are 0-based, SonarQube lines are 1-based
+            return liveIssue;
+        }
+
+        private void BuildProjectPathToIdMap()
+        {
+            projectPathToProjectIdMap = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase);
+
+            IVsSolution solution = this.serviceProvider.GetService<SVsSolution, IVsSolution>();
+            Debug.Assert(solution != null, "Cannot find SVsSolution");
+
+            // TODO: handle return codes
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)